### PR TITLE
Updated pyflexplot to Python 3.10.12

### PR DESCRIPTION
**Pull request by Karko93**
Updated the code, so no error in tests occurs:
- adapted type comparison
- Removed deprecated properties
- Updated references in the slow test of the models

### CODE REVIEW

File: src/pyflexplot/plots.py [hunk 1]
```diff
@@ -902,7 +902,10 @@ def cmap2colors(
             n_colors += 1
         if extend in ["max", "both"] and not color_over:
             n_colors += 1
-        cmap = mpl.cm.get_cmap(cmap, lut=n_colors)
+        cmap = mpl.pyplot.get_cmap(cmap)
+        if n_colors > 1:
+            cmap = mpl.colors.ListedColormap(cmap(np.linspace(0, 1, n_colors)))
```
Comment (cfkanesan) [inline src/pyflexplot/plots.py:907]: `mpl.pyplot.get_cmap` has the `lut` argument so why not use it?

File: src/pyflexplot/plotting/coord_trans.py [hunk 2]
```diff
@@ -133,20 +133,24 @@ def axes_to_map(self, x, y):
...
         if (
             isinstance(self.proj_map, PlateCarree)
-            and not isinstance(x, float)
+            and not isinstance(x_map, float)
             and len(x) == 2
-            and np.isclose(x[0], x[1], atol=0.5)
-            and np.isclose(np.abs(x[0]), 180, atol=0.5)
         ):
-            # Fix edge case where lon range (-180, 180) is turned into (-180, -180)
+            # Fix edge case where lon range (-180, 180) can invert to (180, -180)
```
Comment (cfkanesan) [inline src/pyflexplot/plotting/coord_trans.py:143]: Are there any test cases that expose this behaviour?

File: src/srutils/dataclasses.py [hunk 3]
```diff
@@ -299,45 +302,43 @@ def prepare_wrapped_value(
...
                 pass
-        raise error(value, type_, f"no compatible inner type: {inner_types}")
+        raise error(value, type_name, f"no compatible inner type: {inner_types}")
 
-    elif type_.startswith("typing.Optional["):
+    elif "Optional" in type_name:
         if value in [None, "None"]:
             return None
-        inner_type = type_[len("typing.Optional[") : -len("]")]
+        inner_type = [i for i in get_args(type_) if not i == type(None)][0]
```
Comment (cfkanesan) [inline src/srutils/dataclasses.py:334]: If a single type is expected, the tuple unpacking mechanism can be used to verify this:
```suggestion
        [inner_type] = (i for i in get_args(type_) if not i == type(None))
```

**Review by cfkanesan — COMMENTED** [review]
The changes look okay from my understanding with limited context. The `dataclasses.cast_value` seems like it's providing some functionality that would be best implemented in a dedicated lib such as pydantic or marshmallow. I can see that the former was previously implemented. Any idea why it was then removed? @pirmink

**Comment by pirmink** [comment]
> The dataclasses.cast_value seems like it's providing some functionality that would be best implemented in a dedicated lib such as pydantic or marshmallow. I can see that the former was previously implemented. Any idea why it was then removed?

When I started the update towards Python 3.10, some of the typing-related code created problems, which @twicki solved for me. Maybe it was removed by him in this process and he can answer your question?

**Comment by cfkanesan** [comment]
The removal of pydantic was done about 3 years ago. The commit doesn't reference a PR nor a Jira ticket so it's a bit difficult to establish the reason for this action.

**Comment by ruestefa** [comment]
Yes, I moved away from Pydantic primarily because it was a pain to decipher the exceptions when something went wrong in the code. Quite often, figuring out what ultimately triggered an error in Pydantic devolved into a guessing game, and while I got slightly better at it over time, I just grew tired of it and the overhead just outweighed the benefits. (Maybe Pydantic has improved in this respect in the time since; I haven't used it anymore since then.) That's why there's surely some functionality in the code that could be replaced by Pydantic.

File: src/pyflexplot/plotting/coord_trans.py (hunk 2 above)
Comment (Karko93) [reply src/pyflexplot/plotting/coord_trans.py:143]: Yes, there is one test which covers this edge case

**Review by cfkanesan — APPROVED** [review]
